### PR TITLE
Document and clean up SOAP client types and utility functions

### PR DESCRIPTION
**Pull request by taylorreece**
- Document SOAP client types and utility functions
- Clean up some function overloading
- Ensure SOAP components can be tested on Windows systems

### CODE REVIEW

File: src/clients/soap/utils.ts [hunk 1]
```diff
@@ -20,27 +17,44 @@ import { createClientAsync, Client, SoapMethodAsync } from "soap";
 import axios from "axios";
 import { writeFile, rm } from "fs/promises";
 import { v4 as uuid } from "uuid";
+import os from "os";
+import path from "path";
 
+/**
+ *
+ * @param client
+ */
```
Comment (ryanwersal) [inline src/clients/soap/utils.ts:26]: This intended to be filled in or removed?

File: src/clients/soap/utils.ts [hunk 2]
```diff
@@ -96,16 +124,19 @@ const overrideClientDefaults = (
   }
 };
 
+/**
+ * Make a request to a SOAP-based API
+ * @param param0 An object containing how to connect to a SOAP-based API, the method to onvoke, any overridden values for the HTTP client, and whether or not to log debug messages
```
Comment (ryanwersal) [inline src/clients/soap/utils.ts:129]: Nit: `onvoke` -> `invoke`

Comment (ryanwersal) [inline src/clients/soap/utils.ts:129]: Would it be better to move this part of the docs to the `RequestParams` type itself? I think that would provide better in-editor context though I'm unsure about the actual doc generation tooling.

**Review by ryanwersal — APPROVED** [review]
LGTM 👍

**Review by hipplej — APPROVED** [review]
LGTM

File: src/clients/soap/utils.ts (hunk 2 above)
Comment (taylorreece) [reply src/clients/soap/utils.ts:129]: I think you're right - you can command-click the argument to get more detailed info about each destructured element.

**Pull request merged**